webui/react/src/pages/JobQueue/ManageJob.tsx

```diff
@@ -0,0 +1,146 @@
+import { Form, FormInstance, Input, Modal, Select } from 'antd';
+import React, { useCallback, useRef } from 'react';
+
+import Json from 'components/Json';
+import { useStore } from 'contexts/Store';
+import handleError, { ErrorType } from 'ErrorHandler';
+import { V1SchedulerType } from 'services/api-ts-sdk';
+import { detApi } from 'services/apiConfig';
+import { Job, RPStats } from 'types';
+import { truncate } from 'utils/string';
+
+import RPStatsOverview from './RPStats';
+
+const { Option } = Select;
+
+interface Props {
+  job: Job;
+  onFinish: () => void;
```

Comment:
**nit:** thoughts on making this `optional`? most event handlers tend to follow that pattern but there are exceptions where the component is useless without specific handlers defined.

Reply:
Are you thinking this component might be reused? I made it optional just in case.